addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -0,0 +1,237 @@
+    lead_total_credit = fields.Integer(compute='_compute_lead_total_credit', readonly=True)
+    contact_credit_per_company = fields.Integer(compute='_compute_contact_credit_per_company', readonly=True)
+    contact_total_credit = fields.Integer(compute='_compute_contact_total_credit', readonly=True)
+
+    _sql_constraints = [
+        ('limit_contact_number', 'check(contact_number >= 1 and contact_number <= %d)' % MAX_CONTACT, 'Maximum %d contacts are allowed!' % MAX_CONTACT),
+        ('limit_lead_number', 'check(lead_number >= 1)', 'You need to request at least one company!'),
+        ('check_company_size', 'check(company_size_min <= company_size_max)', 'The minimum size of the company should be less than or equal to the maximum size.'),
+    ]
+
+    @api.depends('lead_number')
+    def _compute_lead_total_credit(self):
+        for req in self:
+            req.lead_total_credit = CREDIT_PER_COMPANY * req.lead_number
+
+    @api.depends('contact_number')
+    def _compute_contact_credit_per_company(self):
+        for req in self:
+            req.contact_credit_per_company = CREDIT_PER_CONTACT * req.contact_number
+
+    @api.depends('contact_number', 'lead_number')
+    def _compute_contact_total_credit(self):
+        for req in self:
+            req.contact_total_credit = CREDIT_PER_CONTACT * req.contact_number * req.lead_number
+
+    def _compute_leads_count(self):
+        leads = self.env['crm.lead'].read_group([
+            ('lead_mining_request_id', 'in', self.ids)
+        ], fields=['lead_mining_request_id', 'type'], groupby=['lead_mining_request_id', 'type'], lazy=False)
+        mapping = {(lead['lead_mining_request_id'][0], lead['type']): lead['__count'] for lead in leads}
+        for request in self:
+            request.leads_count = mapping.get((request.id, 'lead'), 0)
+            request.opportunity_count = mapping.get((request.id, 'opportunity'), 0)
+
+    @api.onchange('lead_number')
+    def _onchange_lead_number(self):
+        if self.lead_number <= 0:
+            self.lead_number = 1
+
+    @api.onchange('contact_number')
+    def _onchange_contact_number(self):
+        if self.contact_number <= 0:
+            self.contact_number = 1
+        elif self.contact_number > MAX_CONTACT:
+            self.contact_number = MAX_CONTACT
+
+    @api.onchange('country_ids')
+    def _onchange_country_ids(self):
+        self.state_ids = []
+
+    @api.multi
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        if default is not None:
+            default['name'] = _('New')
+        else:
+            default = {'name': _('New')}
+        return super(CRMLeadMiningRequest, self).copy(default=default)
+
+    @api.model
+    def default_get(self, fields):
+        res = super(CRMLeadMiningRequest, self).default_get(fields)
+        if not self.env.user.has_group('crm.group_use_lead'):
```

Comment:
I think it is not needed, you can create a default method like:
`def default_lead_type:
  if self.env.user.has_group('crm.group_use_lead'):
    return 'lead'
  else:
    return 'opportunity'
`

et sur le field default=default_lead_type